### ISSUE TITLE
example/user/tool does not work

### ISSUE THREAD

**Issue by jimav** [PerlAlien/Alien-Build#401]
The example code in `example/user/tool/LZMA/Example.pm` does not actually work.

It *appears* to work because the 'xz' program is already installed in my system.  If I temporarily rename /usr/bin/xz then the example's `lzma_version_string` function returns undef.  

I have Alien::xz v0.08 installed.

Most likely the problem is obvious; but I am newly trying to figure out Alien::Build and so don't know where to start.   It would be helpful if the examples could be tested.   Thanks.

```
#!/usr/bin/env perl
use strict; use warnings;

use File::Which;
BEGIN{
  if (my $path = File::Which::which("xz")) {
    die "$path exists;\n",
        "  You must rename or move it aside temporarily for this test\n";
  }
}

use lib "/home/jima/src/Alien-Build-2.79/example/user/tool/lib";
use LZMA::Example qw(lzma_version_string);

my $s = lzma_version_string();
die "Result is undef" unless defined($s);  # **DIES HERE**

print "GOT: $s\n";
```

**Comment by plicease** [comment]
This is expected if `Alien::xz` detected the system xz when it was installed.  If you force a share install (which will build xz) then you should get the built xz

```
$ env ALIEN_INSTALL_TYPE=share cpanm --reinstall Alien::xz
```

**Comment by plicease** [comment]
...but... you are also right there is a bug in the example .pm file.

**Comment by jimav** [comment]
> env ALIEN_INSTALL_TYPE=share cpanm --reinstall Alien::xz

That didn't seem to help:
```
$ env ALIEN_INSTALL_TYPE=share cpanm --reinstall Alien::xz
--> Working on Alien::xz
Fetching http://www.cpan.org/authors/id/P/PL/PLICEASE/Alien-xz-0.08.tar.gz ... OK
Configuring Alien-xz-0.08 ... OK
Building and testing Alien-xz-0.08 ... OK
Successfully reinstalled Alien-xz-0.08
1 distribution installed
$ sudo mv /usr/bin/xz /usr/bin/xz.SAV
$ /tmp/test # the script in the original post
Result is undef at /tmp/test line 16.
```

**Comment by plicease** [comment]
I believe this to be fixed in 2.80. Please let me know if I missed something.  Thank you for raising this issue!
